data/femnist/preprocess/data_to_json.py

```diff
@@ -73,7 +73,7 @@ def relabel_class(c):
         user_data[w]['y'].append(nc)
 
     writer_count += 1
-    if writer_count == MAX_WRITERS:
+    if writer_count == MAX_WRITERS or writer_count == len(writers):
```

Comment:
This change doesn't seem to be enough. Because writer_count gets restarted every MAX_WRITERS, the second condition won't ever be satisfied. I believe you need a counter for all writers that have been dumped.